doc/api/fs.md

```diff
@@ -4446,6 +4451,22 @@ It is unsafe to use `fs.writeFile()` multiple times on the same file without
 waiting for the callback. For this scenario, [`fs.createWriteStream()`][] is
 recommended.
 
+It is possible to abort a `writeFile` with an `AbortSignal`:
+
+```js
+const controller = new AbortController();
+const { signal } = controller;
+const data = new Uint8Array(Buffer.from('Hello Node.js'));
+fs.writeFile('message.txt', data, { signal }, (err) => {
+  // When a request is aborted - the callback is called with an AbortError
+});
+// When the request should be aborted
+controller.abort();
+```
+
+Aborting an ongoing request does not abort individual operating
+system requests but rather the internal buffering `fs.writeFile` performs.
```

Comment:
Users who are unfamiliar with how writeFile works under the covers may find this sentence confusing.

Reply:
How do you propose we deal with this? Add an extra note to writeFile explaining how it works and suggesting a createWriteStream for performacne sensitive stuff like we do with readFile?

Reply:
I think it's better to explain how this works underneath instead, and then include the above sentence.

Reply:
@mcollina @jasnell PTAL